contracts/token/Token.sol

```diff
@@ -154,6 +154,7 @@ contract Token is IERC20 {
     function _transfer(address sender, address recipient, uint256 amount) internal {
         require(sender != address(0), "ERC20: transfer from the zero address");
         require(recipient != address(0), "ERC20: transfer to the zero address");
+        require(amount <= type(uint128).max, "ERC20: transfer amount exceeds uint128::max");
```

Comment:
The precision of ACA is 12, which is 18 in the contract, so it cannot satisfy u128::max.
Although u128::max is unlikely to happen.

Reply:
@zjb0807 in the current design you can pass large number and precompile will ignore first 32 bytes and resulting into a successful transaction but the amount is not the one passed

Reply:
I removed this and will handle it on input hander